client/protocols/grading.py

```diff
@@ -39,15 +39,31 @@ def run(self, messages):
         locked = 0
 
         analytics = {}
+        # check if analytics info is in messages
+        if 'analytics' in messages:
+            started = messages['analytics']['started']
+        else:
+            started = None
 
         for test in self.assignment.specified_tests:
-            log.info('Running tests for {}'.format(test.name))
-            results = test.run()
+            print('Check if tests for {} need to run'.format(test.name))
```

Comment:
This line isn't necessary. Just do the check silently. If it turns out tests do not need to run, we just need to print a line saying so.